config/env_blank.hjson

```diff
@@ -25,7 +25,12 @@
         "API_CLIENT_SECRET": "",
         # Canvas API
         "CANVAS_URL": "",
-        "CANVAS_TOKEN": ""
+        "CANVAS_TOKEN": "",
+        # For processing canvas_placements
+        # ID in each object can be duplicated if the same tool is duplicated
+        "SUPPORTED_LTI_TOOLS" : {
+            "context_external_tool_12345": {"name": "Zoom", "id": 0},
+        }
```

Comment:
This looks like it will work, but if we're going to start doing something with the `external_tool` ID, maybe it just makes sense to start using that ID instead of the `0` and `1`. 

I'd also like to try to keep the configuration as simple as possible. `context_external_tool` isn't going to change right? Maybe this could be condensed to something like the following (the key and value could also be flipped):

```
        "SUPPORTED_LTI_TOOLS" : {
            "ZOOM": 12345
        }
```

Thoughts?

Reply:
I would say it "probably" won't change but I don't know for sure. This is how it's reported in the api call. 

I debating between splitting off the numeric id and just using this numeric id instead of the full string
`id = s.rsplit('_', 1)[1]`

or just putting the full string here. 

I think having an internal id would still need to be something we have because I can imagine multiple context_tools might be the same tool we want to report on. Like 12345 and 51231 could both be Zoom. So I figured we'd use the Canvas ID as the key here.

Reply:
If the nature of the relationship between LTI tool (a Zoom LTI) and LTI provider (Zoom the company) changes from 1:1 to M:1, aren't we going to want to record or identify the separate LTI tools for each service? I dunno, it just seems like using this ID is going to be more precise than letting the configuration classify them (and ultimately hiding differences between them). That's my thinking, but I'll defer to you on this.

If you can remove that repeated string -- or at least make it separately configurable -- I think that would help simplify things.

Reply:
Well an LTI launch is always going to be 1:1. However you can place new tools at different account levels in Canvas (or even possibly at the site level). So we could have a Zoom at the top level and also have a Zoom in LSA. Would we want to call them something different in the config? I don't know? I guess this gives us the flexibility to change the name or use the same identifiers. 

I am going to simplify the string though but I *think* it's possible we may want to just consider multiple ids in Canvas to be the same for our reporting.